### DISCUSSION TITLE
Create table fails with 'schema "_model_draft" does not exist'

### DISCUSSION THREAD

**Discussion by pajot** [openego/oeplatform#889] [Q&A] [answered]
An attempt to create a new table using the wizard in my cloned test installation (launched using uwsgi and nginx) results in the following error:

```
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.InvalidSchemaName) schema "_model_draft" does not exist
LINE 1: CREATE TABLE "_model_draft"."_test_edit" (LIKE "model_draft"...
                     ^

[SQL: CREATE TABLE "_model_draft"."_test_edit" (LIKE "model_draft"."test"INCLUDING ALL EXCLUDING INDEXES, PRIMARY KEY (_id) ) INHERITS (_edit_base);]
(Background on this error at: https://sqlalche.me/e/14/f405)
```

The schema model_draft does exist. I am not sure why it is prefixing model_draft with an underscore.

**Comment by wingechr** [comment, marked as answer]
i think the schema names starting with `_` are for data versioning. if it was not created automatically, you will have to create it manually in the postgres database.

**Comment by pajot** [comment]
I followed the installation instructions exactly. Either they weren't created automatically or there was an error in the attempt.

Do I make _-prefixed versions of all the schemas?

This should be added to the configuration instructions. Do you accept pull requests for the documentation?

Am 8. Februar 2022 07:44:29 MEZ schrieb wingechr ***@***.***>:

**Comment by wingechr** [comment]
no, i think only model_draft. 

I think the current system with separate schemas is scheduled for an overhaul in the future, so I would not put too much work into fixing the docs. But thanks for the info.

**Reply by pajot** [reply]
Is the duplicate schema a duplicate in every way, i.e. _model_draft is structurally identical to model_draft?

**Comment by wingechr** [comment]
you only have to create the schema, nothing else
